docs/adrs/0003-bulk-import-ui.md

```diff
@@ -0,0 +1,98 @@
+* A finite state machine (FSM) that's used to determine cell state.
+* A core system used to render the cell based on state.
+* A message bus that's used to communicate with the rest of the FE application, as well as the BE (kernel) to get job state.
+* An interpreter / manager for app specs - turns the app specs into sets of input form elements.
+* A code translator - turns the various sets of inputs into code that can be executed to run the App.
+
+The reason for the App cell being an extension to code cells is the last point above: in essence, the primary UI components of the App cell serve as a fancy code generator. User interactions with form elements generate the code that can be used to start apps and view their results. Thus, anything that an app cell does with the UI can be done programmatically by an advanced user.
+
+> It should also be noted here that, regardless of the choice made, we may need to alter how app specs are made, interpreted, and rendered. The bulk import design has various inputs and outputs listed in horizontal rows while the existing app specs are rendered vertically. There's no notion of rows/columns in the app spec right now or how to visually display various inputs along a grid pattern. In order to effectively match the design we may need to do one of the following:
+> 1. Alter the app specs to support multiple in-line parameters while retaining backward compatibility (including documentation, etc.).
+> 2. Use a distinct app spec for importers that can include these options.
+> 3. Hard-code how all import apps look and feel in the Narrative interface itself.
+>
+> If we decide to update how app specs work, that means working in the `narrative_method_store` and `catalog` repos as well as internal documentation.
+
+## Alternatives Considered
+
+* Implement the UI in the existing App Cell codebase
+* Implement the UI in a new Cell type
+* Don't use a cell, but some other interface
+
+## Decision Outcome
+
+## Consequences
+
+## Pros and Cons of the Alternatives
+
+### Implement the UI in the existing App Cell code base
+
+This option would entail adding a different mode for the current App Cell, likely based on some additional metadata feature.
+
+* `+` Reduce spread of additional, yet similar, cell types with similar code.
+* `+` Reuse existing cell control mechanisms currently in place with minimal modification.
+* `+` Reuse existing wrapping UI elements - cell title, icons, tabs, etc.
+* `-` Work done to the App cell needs to be made backward compatible with existing functionality.
+* `-` Any "breaking" work (i.e. refactor, etc.) will need to be complete before any releases are possible, making intermediate releases possibly challenging.
+* `-` Adding yet another mode for the App cell will increase the maintenance burden on an already complex code base - see especially the FSM.
+* `-` The draft UI is very complex and may turn the existing code into a deep series of nested logic.
```

Comment:
* `-` Import app cells support re-runability, but with the 90 day file deletion of the staging area and narrative copies, rerunning will eventually fail (true for next option as well)

Reply:
Oof good point - I wonder if design has considered this as well

Reply:
TBF Bill made this point later on in the doc, I just reiterated it here for the app cons